mesonbuild/compilers/compilers.py

```diff
@@ -548,6 +548,22 @@ def to_native(self, copy: bool = False) -> typing.List[str]:
                 # Last occurrence of a library
                 new.insert(group_end + 1, '-Wl,--end-group')
                 new.insert(group_start, '-Wl,--start-group')
+        # Remove system/default include paths added with -isystem
+        if hasattr(self.compiler, 'get_default_include_dirs'):
+            default_dirs = self.compiler.get_default_include_dirs()
+            bad_idx_list = []
+            for i, each in enumerate(new):
+                # Remove the current and and next list element
+                if (each == '-isystem' and
+                        i < (len(new) - 1) and
```

Comment:
Isn't this overly defensive programming? If `-isystem` is the last option we have bigger problems..

Reply:
Here, I would disagree. Yes, a lone `-isystem` will generate a compiler error, but that shouldn't be an excuse for an `IndexError` exception in meson. Finding such a bug should be much easier with a compiler error, rather than a traceback here.